src/components/form/FormFieldVoucher.vue

```diff
@@ -0,0 +1,158 @@
+<script lang="ts">
+/**
+ * FormFieldVoucher Component
+ *
+ * @description * Use this component to render a Voucher widget.
+ * It handles voucher submission and validation and emits
+ * the data about the voucher.
+ *
+ * Used in `RegisterChallengePayment` component.
+ *
+ * @events
+ * - `update:voucher`: Emitted after voucher is successfully applied.
+ *
+ * @components
+ * - `FormFieldTextRequired`: Component to render required text field.
+ *
+ * @example
+ * <form-field-voucher @update:voucher="onUpdateVoucher" />
+ *
+ * @see [Figma Design](https://www.figma.com/design/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?node-id=6410-2305&t=gB7ERmDZorpD4TdE-1)
+ */
+
+// libraries
+import { Notify } from 'quasar';
+import { defineComponent, ref } from 'vue';
+
+// components
+import FormFieldTextRequired from '../global/FormFieldTextRequired.vue';
+
+// composables
+import { i18n } from '../../boot/i18n';
+
+// types
+import type { FormPaymentVoucher } from '../types/Form';
+
+// fixtures
+import voucherFull from '../../../test/cypress/fixtures/registerPaymentVoucherFull.json';
+import voucherHalf from '../../../test/cypress/fixtures/registerPaymentVoucherHalf.json';
+
+export default defineComponent({
+  name: 'FormFieldVoucher',
+  components: {
+    FormFieldTextRequired,
+  },
+  emits: ['remove:voucher', 'update:voucher'],
+  setup(props, { emit }) {
+    const code = ref('');
+    const voucher = ref<FormPaymentVoucher | null>(null);
+
+    /**
+     * Submits voucher data to API
+     * If voucher is valid it emits the data
+     * @returns {void}
+     */
+    const onSubmitVoucher = (): void => {
+      // TODO: Add API call and remove dummy data
+      if (code.value === 'FULL') {
```

Comment:
'FULL' and 'HALF' string literal should be TS Enum.

Reply:
Replaced with enums.